packages/react-stylesheets/src/StylesheetProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { StylesheetContext, StylesheetContextType } from './StylesheetContext';
+import { StylesheetProviderProps } from './StylesheetProvider.types';
+
+/**
+ * Provider for registering stylesheets in a given target document.
+ * The `register` method can be called many times and will only register once
+ * per unique target document.
+ */
+export const StylesheetProvider = (props: React.PropsWithChildren<StylesheetProviderProps>) => {
+  const context = React.useContext(StylesheetContext);
+  const mergedContext = React.useMemo<StylesheetContextType>(
+    () => ({
+      // tslint:disable-next-line:no-any
+      ...(context as any),
+      ...props,
+    }),
+    [context],
```

Comment:
need to check `props` too
i think here should be: `[...context, ...props]`

Reply:
I want the context object to be the key only; you can't do `...props` in an array without first converting to an array using Object.values, which is way less efficient than it should be. I'd rather assume props won't change for the perf benefit of predictably avoiding mutations.

Reply:
i added a comment to clarify though.